src/controllers/hid/hidcontroller.h

```diff
@@ -36,47 +39,21 @@ class HidController final : public Controller {
     int open() override;
     int close() override;
 
-    bool poll() override;
-
   private:
-    bool isPolling() const override;
-    void processInputReport(int bytesRead);
+  signals:
+    void sendOutputReport(const QByteArray& data, unsigned int reportID);
 
+  private:
     // For devices which only support a single report, reportID must be set to
     // 0x0.
     void sendBytes(const QByteArray& data) override;
-    void sendBytesReport(QByteArray data, unsigned int reportID);
-    void sendFeatureReport(const QByteArray& reportData, unsigned int reportID);
-
-    // getInputReport receives an input report on request.
-    // This can be used on startup to initialize the knob positions in Mixxx
-    // to the physical position of the hardware knobs on the controller.
-    // The returned data structure for the input reports is the same
-    // as in the polling functionality (including ReportID in first byte).
-    // The returned list can be used to call the incomingData
-    // function of the common-hid-packet-parser.
-    QByteArray getInputReport(unsigned int reportID);
-
-    // getFeatureReport receives a feature reports on request.
-    // HID doesn't support polling feature reports, therefore this is the
-    // only method to get this information.
-    // Usually, single bits in a feature report need to be set without
-    // changing the other bits. The returned list matches the input
-    // format of sendFeatureReport, allowing it to be read, modified
-    // and sent it back to the controller.
-    QByteArray getFeatureReport(unsigned int reportID);
 
-    const mixxx::hid::DeviceInfo m_deviceInfo;
+    std::shared_ptr<const mixxx::hid::DeviceInfo> m_pDeviceInfo;
```

Comment:
Why don't you simple pass a copy of the data to the thread? This only happens once. I don't see any compelling reason to introduce a nullable shared_ptr to immutable data here.

Reply:
The whole class contains only native types and implicitly shared Qt types that are cheap to copy. And even if copying would require allocations I would prefer an independent copy.

Reply:
This pointer is also used in each instance of HidReport. Creating the pointer in HidController makes it cheap to pass it to this sub class instances.
If you insist I will change it at all places to copy-by-value. But I wonder, if it's really the right approach to have this data dupplication. Do you think passing by reference would be suitable here too?